### DISCUSSION TITLE
Custom Dropdown Toggle

### DISCUSSION THREAD

**Discussion by mdmontesinos** [Megabit/Blazorise#5111] [General]
Is there any way to open a dropdown with a custom toggle instead of the regular one?

Basically, I want to have a Dropdown that is opened using an icon instead of a button, something like this:

![image](https://github.com/Megabit/Blazorise/assets/90258222/4be699a9-5011-4365-9a9f-be9993865d6d)

```html
<Dropdown @ref="_dropdown">
    <DropdownToggle>
        <Icon Name="IconName.Bell"></Icon>
    </DropdownToggle>
    <DropdownMenu>
        <DropdownItem>Item 1</DropdownItem>
        <DropdownItem>Item 2</DropdownItem>
        <DropdownItem>Item 3</DropdownItem>
    </DropdownMenu>
</Dropdown>
```

But I still get the toggle icon, which I would like to get rid of.

I've also tried using the ref and DropdownMenuTargetId to programatically open the dropdown:

```html
<Dropdown @ref="_dropdown" DropdownMenuTargetId="target">
    <DropdownMenu>
        <DropdownItem>Item 1</DropdownItem>
        <DropdownItem>Item 2</DropdownItem>
        <DropdownItem>Item 3</DropdownItem>
    </DropdownMenu>
</Dropdown>

<Button Color="Color.Primary" id="target" @onclick="@(() => ShowMenu())">Toggle</Button>
```

But if I remove the DropdownToggle or set Display to None, the position of the dropdown is sometimes bugged, and also can't be closed by clicking elsewhere.

**Comment by stsrki** [comment]
Just set the `ToggleIconVisible`, eg.

```
<DropdownToggle Color="Color.Secondary" ToggleIconVisible="false">Menu Up</DropdownToggle>
```

**Reply by mdmontesinos** [reply]
I just tried that, but still get the same result.

```html
<Dropdown>
    <DropdownToggle ToggleIconVisible="false">
        <Icon Name="IconName.Bell"></Icon>
    </DropdownToggle>
    <DropdownMenu>
        <DropdownItem>Item 1</DropdownItem>
        <DropdownItem>Item 2</DropdownItem>
        <DropdownItem>Item 3</DropdownItem>
    </DropdownMenu>
</Dropdown>
```

![image](https://github.com/Megabit/Blazorise/assets/90258222/f0adc713-9efa-4a09-88ef-8b74a932ee5f)

**Reply by mdmontesinos** [reply]
Also, I missed that option because the parameter is not shown in the docs (or at least I can't find it).

![image](https://github.com/Megabit/Blazorise/assets/90258222/09283bfa-9a6c-45c8-9fa0-4a7294d7c22c)

**Reply by stsrki** [reply]
That's strange. What provider are you using? BS4 or BS5?

Also, we should add that to the docs.

**Reply by mdmontesinos** [reply]
I'm actually using Tailwind.

**Reply by stsrki** [reply]
That might be a bug. It is going to the issue.

**Reply by mdmontesinos** [reply]
Thanks!

**Reply by mdmontesinos** [reply]
Fixed via #5114
